src/wallets/IdentityWallet.ts

```diff
@@ -0,0 +1,179 @@
+  // TODO: make this class a TLSigner as part of https://github.com/trustlines-network/clientlib/issues/194
+
+  public provider: TLProvider
+
+  private wallet: ethers.Wallet
+  private identityAddress: string
+
+  constructor(provider: TLProvider) {
+    this.provider = provider
+  }
+
+  public get address(): string {
+    return this.identityAddress
+  }
+
+  public get pubKey(): string {
+    // TODO: this makes no sense to me: it is the pubkey of the owner of the identity
+    return this.wallet
+      ? ethers.utils.computePublicKey(this.wallet.privateKey)
+      : undefined
+  }
+
+  public async getAddress(): Promise<string> {
+    if (!this.wallet) {
+      throw new Error('No wallet loaded.')
+    }
+    return this.address
+  }
+
+  /**
+   * Creates a new wallet and encrypts it with the provided password.
+   * @param password Password to encrypt keystore.
+   * @param progressCallback Callback function for encryption progress.
+   */
+  public async createAccount(
+    password: string,
+    progressCallback?: any
+  ): Promise<UserObject> {
+    this.wallet = ethers.Wallet.createRandom()
+    const pubKey = this.wallet.address
+    const encryptedKeystore = await this.wallet.encrypt(
+      password,
+      typeof progressCallback === 'function' && progressCallback
+    )
+
+    const deployIdentityEndpoint = 'identities'
+
+    const identity = await this.provider.fetchPostEndpoint<DeployedIdentity>(
+      deployIdentityEndpoint,
+      this.wallet.address
+    )
+
+    this.identityAddress = identity.identity
+
+    return {
+      address: identity.identity,
+      keystore: encryptedKeystore,
+      pubKey: this.pubKey
+    }
+  }
+
+  /**
+   * Decrypts given keystore and loads wallet.
+   * @param encryptedKeystore Encrypted keystore from `createAccount`.
+   * @param password Password to decrypt keystore.
+   * @param identityAddress the address of the corresponding identity contract
+   * @param progressCallback Callback function for decryption progress.
+   */
+  public async loadAccount(
+    encryptedKeystore: string,
+    password: string,
+    identityAddress: string,
+    progressCallback?: any
+  ): Promise<UserObject> {
+    this.wallet = await ethers.Wallet.fromEncryptedJson(
+      encryptedKeystore,
+      password,
+      typeof progressCallback === 'function' && progressCallback
+    )
+
+    this.identityAddress = identityAddress
+
+    return {
+      address: identityAddress,
+      keystore: encryptedKeystore,
+      pubKey: this.pubKey
+    }
+  }
+
+  /**
+   * Recovers wallet from mnemonic phrase and encrypts keystore with given password.
+   * @param seed Mnemonic seed phrase.
+   * @param password Password to encrypt recovered keystore.
+   * @param identityAddress the address of the corresponding identity contract
+   * @param progressCallback Callback function for encryption progress.
+   */
+  public async recoverFromSeed(
+    seed: string,
+    password: string,
+    identityAddress: string,
```

Comment:
I also dont  know how that makes sense, the user will only have backuped the seed, so this does not help. We should maybe discuss this.